File: gym-backend/src/main/java/com/bequre/pojo/Coach.java
package com.bequre.pojo;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDate;

@Data
@Schema(description = "私教实体")
@TableName("coaches")
@Builder
public class Coach {

    @NotNull
    @Schema(name = "id", description = "ID")
    @TableId(value = "id", type = IdType.AUTO)
    private Integer id;

    @Schema(name = "name", description = "姓名")
    private String name;

    @NotNull
    @Schema(name = "phone", description = "电话")
    private String phone;

    @NotNull
    @Schema(name = "height", description = "身高")
    private Integer height;

    @NotNull
    @Schema(name = "weight", description = "体重")
    private Integer weight;

    @NotNull
    @Schema(name = "joinDate", description = "加入日期")
    private LocalDate joinDate;

    @NotNull
    @Schema(name = "salary", description = "薪水")
    private Integer salary;


}

File: gym-front/src/types/coach.ts
/**
 * Coach，私教实体
 * 某些属性用string是便于前端处理，不用管校验（因为后端已经完善了）
 */
export interface Coach {
    /**
     * ID
     * 添加的请求不需要这个参数
     */
    id?: string;
    /**
     * 姓名
     */
    name: string;
    /**
     * 电话
     */
    phone: string;
    /**
     * 身高
     */
    height: string;
    /**
     * 体重
     */
    weight: string;
    /**
     * 加入日期
     */
    joinDate?: Date;
    /**
     * 薪水
     */
    salary: string;
}

/**
 * 私教分页条件查询实体
 */
export interface CoachQuery {
    /**
     * 姓名
     */
    name?: string | null;

    /**
     * 电话
     */
    phone?: string | null;

    /**
     * 身高
     */
    height?: string | null;

    /**
     * 页码
     */
    pageNo: number;

    /**
     * 每页条数
     */
    pageSize: number;
}

